torchvision/datasets/utils.py

```diff
@@ -24,24 +24,12 @@
 USER_AGENT = "pytorch/vision"
 
 
-def _save_response_content(
-    content: Iterator[bytes],
-    destination: Union[str, pathlib.Path],
-    length: Optional[int] = None,
-) -> None:
-    with open(destination, "wb") as fh, tqdm(total=length) as pbar:
-        for chunk in content:
-            # filter out keep-alive new chunks
-            if not chunk:
-                continue
```

Comment:
The new changes are only equivalent to the old ones if this `continue` was meant to be a `break`. Was this meant to be a `break`? In other words, can there be an empty chunk and then a non-empty chunk?

Reply:
TL;DR they are equivalent since `content` now can only be one type or iterator for which the branch will never trigger.

These keep-alive chunks are `b""`. Content iterators coming from `requests` could actually contain them. However, the content iterator coming from our `_urlretrieve` function cannot. 

The way `iter` works when you have two inputs is quite unintuitive. Basically waht `iter(lambda: response.read(chunk_size), b"")` does is it calls the `lambda` forever until that returns the sentinel `b""`. If that happens it raises a `StopIteration`. Meaning

```python
for chunk in iter(lambda: response.read(chunk_size), b""):
    if not chunk:
        # this will never be reached
```

Reply:
To answer your question directly:

> In other words, can there be an empty chunk and then a non-empty chunk?

No.

```python
items = [b"foo", b"", b"bar"]
idx = 0


def fn():
    global idx

    item = items[idx]
    idx += 1
    return item


for chunk in iter(fn, b""):
    print(chunk)
```

```
b'foo'
```